conjure_oxide/src/utils/conjure.rs

```diff
@@ -174,8 +174,8 @@ pub fn get_solutions_from_conjure(
         for (name, value) in solution {
             let name = Name::UserName(name.to_owned());
             let value = match value {
-                JsonValue::Bool(b) => Ok(Constant::Bool(*b)),
-                JsonValue::Number(n) => Ok(Constant::Int(n.as_i64().unwrap().try_into().unwrap())),
+                JsonValue::Bool(b) => Ok(Literal::BoolL(*b)),
```

Comment:
constructors are already namespaced under the type name, so why not Literal::Bool?

Reply:
I am doing `use Expr::*` and `use Factor::*` in lots of the code, which causes conflicts between `Factor` the expression and `Factor` the type.

(this might be an anti-pattern, so I'll be happy to change it)

Reply:
I think it is probably an anti-pattern, at least as far as I know. Let's stick to Bool/Int etc and Literal/Reference

Reply:
I've kept  Expression:::FactorE as we import * this everywhere and clippy doesn't know how to fix it. The others I've removed though 

Before we merge this I'll have another go at linting this

Reply:
I think fixing Expr::* everywhere would be too annoying, so if we are ok merging it with FactorE, let's do thar

Reply:
Was using the wrong lint!

Reply:
The lint clippy::enum_glob_use works.

However, the fix it gave was useless...

![Screenshot 2024-11-08 at 09 45 04](https://github.com/user-attachments/assets/f692a640-6ba8-4f7b-9c74-5dde1c370aac)